api/authorizations.ts

```diff
@@ -1,144 +1,141 @@
-						if (error.indexOf("Unique constraint failed on the constraint: `authorization`") > -1)
+						const idUnit = parseInt(req.query.id_unit as string);
+						if (!idUnit) return res.status(404).json({ Message: "missing <id_unit>" });
+
+						if (!req.query.room_ids) return res.status(404).json({ Message: "missing <room_ids> list of lab ids" });
+						const roomIds = (req.query.room_ids as string).split(',');
+
+						if (!req.query.scipers) return res.status(404).json({ Message: "missing <scipers> list of authorisation holders" });
+						const scipers = (req.query.scipers as string).split(',');
+
+						const cas = (req.query.cas as string).split(',');
+						const args = {
+							id_unit: idUnit,
+							authorization: request,
+							creation_date: (new Date()).toLocaleDateString("en-GB"),
+							expiration_date: (new Date(expirationDate)).toLocaleDateString("en-GB"),
+							status: "Active",
+							type: "Chemical",
+							cas: [
+								cas.map(c => `{name: "${c}", status: "New"}`)
+							],
+							holders: [
+								scipers.map(sc => `{sciper: ${sc}, status: "New"}`)
+							],
+							rooms: [
+								roomIds.map(r => `{id: ${r}, status: "New"}`)
+							],
+						}
+						const add = await addAuthorization(args, context);
+						if (add.isSuccess)
 							res.json({Message: "Ok"});
-						else
-							res.json({Message: error});
+						else {
+							const error = add.errors.map(err => err.message).join(', ');
+							if (error.indexOf("Unique constraint failed on the constraint: `authorization`") > -1)
+								res.json({Message: "Ok"});
+							else
+								res.json({Message: error});
+						}
 					}
 					break;
 				case "auth_renew":
-					const reqParts = request.split("-");
-					const requestNumber = `${reqParts[0]}-${reqParts[1]}`;
-					const queryForAuth = `query fetchChemicalAuthorizations {
-							authorizationsWithPagination (take: 0, skip: 0, search: "Authorization=${requestNumber}", type: "Chemical", token: "${token}") {
-								authorizations{
-									id
-								}
-								totalCount
-							}
-						}`;
-					const resultForAuth = await makeQuery(queryForAuth, 'SNOW');
-					if (resultForAuth.authorizationsWithPagination.totalCount === 1) {
-						query = `mutation updateRadioprotection {
-								updateAuthorization(
-									token: "${token}",
-									id: ${JSON.stringify(resultForAuth.authorizationsWithPagination.authorizations[0].id)},
-									expiration_date: "${(new Date(expirationDate)).toLocaleDateString("en-GB")}",
-									status: "Active",
-									renewals: ${parseInt(reqParts[2])}
-								)
-							 {
-								errors {
-									message
-								}
-								isSuccess
-							}
-						}`;
-					} else {
-						return res.status(404).json({ Message: "Could not find parent authorisation" });
+					if ( !req.user.canCallAPIToRenewAuthorization )
+						res.status(403).json({Message: 'Unauthorized'});
+					else {
```

Comment:
Please don't do long-standing `else` blocks like this; rather, `return` early in the `if` branch.